src/gt4py/next/iterator/transforms/global_tmps.py

```diff
@@ -512,13 +561,22 @@ def update_domains(
                             (axis, range_) if axis != old_axis else (new_axis, new_range)
                             for axis, range_ in consumed_domain.ranges.items()
                         )
+                        # TODO(tehrengruber): Revisit. Somehow the order matters so preserve it.
```

Comment:
for `unstructured_domain(horizontal, vertical)` the order matters, is that the point here?

Reply:
I'm guessing that it is more the `TemporaryAllocation` where the order matters, but I am not sure.